crates/meilisearch-auth/src/store.rs

```diff
@@ -82,45 +81,15 @@ impl HeedAuthStore {
         // Delete existing actions
         self.delete_key_from_inverted_db(&mut wtxn, &uid)?;
 
-        // Create new actions with bitflags
-        let mut actions = HashSet::new();
-        for action in &key.actions {
-            match *action {
-                Action::All => actions.extend(enum_iterator::all::<Action>()),
-                Action::DocumentsAll => {
-                    actions.extend(
-                        [Action::DocumentsGet, Action::DocumentsDelete, Action::DocumentsAdd]
-                            .iter().copied(),
-                    );
-                }
-                Action::IndexesAll => {
-                    actions.extend(
-                        [
-                            Action::IndexesAdd,
-                            Action::IndexesDelete,
-                            Action::IndexesGet,
-                            Action::IndexesUpdate,
-                            Action::IndexesSwap,
-                        ]
-                        .iter().copied(),
-                    );
-                }
-                Action::SettingsAll => {
-                    actions.extend([Action::SettingsGet, Action::SettingsUpdate].iter().copied());
-                }
-                Action::TasksAll => {
-                    actions.extend([Action::TasksGet, Action::TasksDelete, Action::TasksCancel].iter().copied());
-                }
-                other => {
-                    actions.insert(other);
-                }
-            }
-        }
+        //Create a new Bit Mask
 
         // Store the actions with bitflags
-        let key_actions = KeyActions::new(&actions, &key.indexes, key.expires_at);
-        self.key_actions.put(&mut wtxn, uid.as_bytes(), &key_actions)?;
-
+        let mut bit_mask = 0;
+        for action in &key.actions {
+            bit_mask |= action.bits();
+        }
+        let key_masks = KeyMasks::new(bit_mask, &key.indexes, key.expires_at);
```

Comment:
This looks unnecessary, because it's the same logic as what is done in the `KeyMasks::new`.

```suggestion
        let key_masks = KeyMasks::new(key);
```